src/components/products/product-list-item/product-list-item.tsx

```diff
@@ -0,0 +1,20 @@
+import { Link } from "react-router-dom";
+import './product-list-item.scss'
+import { Product } from "../../../interfaces/products/product.interface";
+
+interface ProductProps {
+  product: Product;
+}
+
+export default function ProductListItem({product}: ProductProps) {
+  return (
+    <>
+      <tr>
+            <td>{product.category.name}</td>
+            <td>{product.name}</td>
+            <td>{product.price}</td>
+            <td className= "detail-link-col"><Link to={`/products/${product.id}`}><button className="detail-page-btn">{">"}</button></Link></td>
```

Comment:
Here I think we can improve just a little the app.
Let's say we are on /products page. We can go on the page of a specific product in 2 ways: 
- by clicking on the product in the table
- by changing the URL to /products/c49e49f6-ee3f-4897-a75c-53e9f9152248 (let's assume I know this from my head)

Right now you navigate to that page and then fetch the product from the backend.
But what we could do is:
- if the user clicks on the product in the table we send it as props because we have all details about the product (I think): ````navigate(`/products/${product.id}`, { state: { product } });````
- if the user changes the URL or refreshes the page we do a call to backend to get the details.

In this way we do less calls to backend. This change probably requires some changes :) **I wanted to tell you about this but implement this only if you have time, is not mandatory**